applications/tests/test_applications_mutations.py

```diff
@@ -702,3 +717,315 @@ def test_delete_winter_storage_application_inexistent_application(
+            acceptOtherCultureNews
+            acceptBoatingNewsletter
+            winterStorageAreaChoices {
+                winterStorageArea {
+                    id
+                }
+            }
+        }
+    }
+}
+"""
+
+
+def test_update_winter_storage_application_by_owner(
+    berth_customer_api_client, winter_storage_application, customer_profile
+):
+    winter_storage_application_id = to_global_id(
+        WinterStorageApplicationNode, winter_storage_application.id
+    )
+    remove_choice = WinterAreaChoiceFactory(application=winter_storage_application)
+
+    place = WinterStoragePlaceFactory()
+    area_node_id = to_global_id(
+        WinterStorageAreaNode, place.winter_storage_section.area.id
+    )
+
+    customer_profile.user = berth_customer_api_client.user
+    customer_profile.save()
+    winter_storage_application.customer = customer_profile
+    winter_storage_application.save()
+
+    variables = {
+        "id": winter_storage_application_id,
+        "language": "en",
+        "firstName": "John",
+        "lastName": "Doe",
+        "phoneNumber": "1234567890",
+        "email": "john.doe@example.com",
+        "address": "Mannerheimintie 1",
+        "zipCode": "00100",
+        "municipality": "Helsinki",
+        "boatWidth": Decimal("2.00"),
+        "boatLength": Decimal("3.00"),
+        "informationAccuracyConfirmed": True,
+        "acceptFitnessNews": False,
+        "acceptLibraryNews": False,
+        "acceptOtherCultureNews": False,
+        "acceptBoatingNewsletter": True,
+        "addChoices": [{"winterAreaId": area_node_id, "priority": 1}],
+        "removeChoices": [to_global_id(WinterStorageAreaChoiceType, remove_choice.id)],
+    }
+
+    assert winter_storage_application.winterstorageareachoice_set.count() == 1
+
+    executed = berth_customer_api_client.execute(
+        UPDATE_WINTER_STORAGE_APPLICATION_OWNER_MUTATION, input=variables
+    )
+
+    assert executed == {
+        "data": {
+            "updateWinterStorageApplication": {
+                "winterStorageApplication": {
+                    "id": winter_storage_application_id,
+                    "language": "EN",
+                    "firstName": "John",
+                    "lastName": "Doe",
+                    "phoneNumber": "1234567890",
+                    "email": "john.doe@example.com",
+                    "address": "Mannerheimintie 1",
+                    "zipCode": "00100",
+                    "municipality": "Helsinki",
+                    "boatWidth": "2.00",
+                    "boatLength": "3.00",
+                    "informationAccuracyConfirmed": True,
+                    "acceptFitnessNews": False,
+                    "acceptLibraryNews": False,
+                    "acceptOtherCultureNews": False,
+                    "acceptBoatingNewsletter": True,
+                    "winterStorageAreaChoices": [
+                        {"winterStorageArea": {"id": area_node_id}}
+                    ],
```

Comment:
Trivial thing, but this could be a merge of the "variables" -variable and the changed entries so it would be easy to see which values are changed https://stackoverflow.com/a/26853961/784642